src/libs/kata-sys-util/src/mount.rs

```diff
@@ -0,0 +1,1087 @@
+const PROC_FIELDS_PER_LINE: usize = 6;
+const PROC_DEVICE_INDEX: usize = 0;
+const PROC_PATH_INDEX: usize = 1;
+const PROC_TYPE_INDEX: usize = 2;
+
+// Sadly nix/libc doesn't have UMOUNT_NOFOLLOW although it's there since Linux 2.6.34
+const UMOUNT_NOFOLLOW: i32 = 0x8;
+
+lazy_static! {
+    static ref MAX_MOUNT_PARAM_SIZE: usize =
+        if let Ok(Some(v)) = unistd::sysconf(unistd::SysconfVar::PAGE_SIZE) {
+            v as usize
+        } else {
+            4096
+        };
+}
+
+/// Errors related to filesystem mount operations.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Can not bind mount {0} to {1}: {2}")]
+    BindMount(PathBuf, PathBuf, nix::Error),
+    #[error("Failure injection: {0}")]
+    FailureInject(String),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error("Invalid mountpoint entry (expected {0} fields, got {1}) fields: {2}")]
+    InvalidMountEntry(usize, usize, String),
+    #[error("Invalid mount option: {0}")]
+    InvalidMountOption(String),
+    #[error("Invalid path: {0}")]
+    InvalidPath(PathBuf),
+    #[error("Failure in waiting for thread: {0}")]
+    Join(String),
+    #[error("Can not mount {0} to {1}: {2}")]
+    Mount(PathBuf, PathBuf, nix::Error),
+    #[error("Mount option exceeds 4K size")]
+    MountOptionTooBig,
+    #[error("Path for mountpoint is null")]
+    NullMountPointPath,
+    #[error("Faile to open file {0} by path, {1}")]
+    OpenByPath(PathBuf, io::Error),
+    #[error("Can not read metadata of {0}, {1}")]
+    ReadMetadata(PathBuf, io::Error),
+    #[error("Can not remount {0}: {1}")]
+    Remount(PathBuf, nix::Error),
+    #[error("Can not find mountpoint for {0}")]
+    NoMountEntry(String),
+    #[error("Can not umount {0}, {1}")]
+    Umount(PathBuf, io::Error),
+}
+
+/// A specialized version of `std::result::Result` for mount operations.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Information of mount record from `/proc/mounts`.
+pub struct LinuxMountInfo {
+    /// Source of mount, first field of records from `/proc/mounts`.
+    pub device: String,
+    /// Destination of mount, second field of records from `/proc/mounts`.
+    pub path: String,
+    /// Filesystem type of mount, third field of records from `/proc/mounts`.
+    pub fs_type: String,
+}
+
+/// Get the device and file system type of a mount point by parsing `/proc/mounts`.
+pub fn get_linux_mount_info(mount_point: &str) -> Result<LinuxMountInfo> {
+    let mount_file = fs::File::open(PROC_MOUNTS_FILE)?;
+    let lines = io::BufReader::new(mount_file).lines();
+
+    for mount in lines.flatten() {
+        let fields: Vec<&str> = mount.split(' ').collect();
+
+        if fields.len() != PROC_FIELDS_PER_LINE {
+            return Err(Error::InvalidMountEntry(
+                PROC_FIELDS_PER_LINE,
+                fields.len(),
+                mount,
+            ));
+        }
+
+        if mount_point == fields[PROC_PATH_INDEX] {
```

Comment:
Nit: Maintenance/safety: I can see this is correct, but it would still be safer to use `.get()` as that's checked in the case that `PROC_FIELDS_PER_LINE` changed (unlikely though that is :)